js/build.js

```diff
@@ -102,6 +102,18 @@
               return response;
             }
 
+            if (data.dataSourceQuery.selectedModeIdx === 1) {
+              // Add aggregate query options for Summary mode
+              _.set(data, 'dataSourceQuery.query.aggregate', [
+                {
+                  $group: {
+                    _id: '$data.' + data.dataSourceQuery.columns.column,
```

Comment:
you checked whether this works fine with column names such as `Foo bar '#!`?

Reply:
I'll confirm

Reply:
@squallstar It works pretty well with spaces and other odd characters.

**Request**

![image](https://user-images.githubusercontent.com/290733/76554752-4ab91a80-648e-11ea-89cf-65e17c06d033.png)

**Response**

![image](https://user-images.githubusercontent.com/290733/76554770-54db1900-648e-11ea-99b2-b5d286dfa273.png)

However, it doesn't work with the column name contains a dot `.`.

**Request**

![image](https://user-images.githubusercontent.com/290733/76555096-e3e83100-648e-11ea-9de8-a94c5d826710.png)

**Response**

![image](https://user-images.githubusercontent.com/290733/76555111-ecd90280-648e-11ea-8f03-d8752ed24e18.png)

What do you recommend?

Reply:
https://stackoverflow.com/questions/12397118/mongodb-dot-in-key-name tells me there's no solution, but I wonder if mingo has its own solution.

Reply:
I've tried the following options and none of them worked.

- `_id: ['$data', data.dataSourceQuery.columns.column]`
- `_id: '$data.[' + data.dataSourceQuery.columns.column + ']'`
- `_id: '$data."' + data.dataSourceQuery.columns.column + '"'`

Reply:
@tonytlwu I assume the logic in place prior to your changes was fine with columns having dots in it? To be honest I do accept it as a limitation since dots in column names are extremely unlikely. As a matter of fact I've never seen anyone using it.

The alternatives are:
- you keep the legacy logic if a column has a dot in its name
- we investigate on whether there's a way to escape the dot when querying on Mingo

Reply:
> you keep the legacy logic if a column has a dot in its name

@squallstar I was afraid it'd come to this too.

I quickly checked the Mingo source code and it looks like it's simply doing a `.split('.')` so we'd have to do this.

I've committed a tested update so you can review it all again now. Once you're happy with it I'll update the other charts too.